### PR TITLE
Added VirtualNode support

### PR DESCRIPTION
**Pull request by shunr-hpe**
### Summary and Scope

Added support for the VirtualNode type

### Issues and Related PRs

* Resolves [CASM-XYZ](https://jira-pro.it.hpe.com:8443/browse/CASMHMS-6071)

### Testing

Tested on:

* development system

Were the install/upgrade based validation checks/tests run?(goss tests/install-validation doc)
Were continuous integration tests run? Y/N   If not, Why? Y
Was an Upgrade tested?                 Y/N   If not, Why? N
Was a Downgrade tested?                Y/N   If not, Why? N

### CODE REVIEW

File: test/ct/api/2-disruptive/test_hardware.tavern.yaml [hunk 1]
```diff
@@ -267,6 +267,60 @@ stages:
     response:
       status_code: 201
 
+  - name: slsAPIHardwarePUTNewVirtualNode - Create a new entry VirtualNode with put
```
Comment (rsjostrand-hpe) [inline test/ct/api/2-disruptive/test_hardware.tavern.yaml:270]: Maybe add a test with post

**Comment by shunr-hpe** [comment]
Here is the diff between the old `<` and new `>` entries in the xname pattern
```
> ^d([0-9]+)w([0-9]+)$
7a9
> ^x([0-9]{1,4})m([0-3])p([0-7])v([1-9][0-9]*)$
12a15
> ^x([0-9]{1,4})c([0-7])b([0])i([0-3])$
14c17
< ^x([0-9]{1,4})c([0-7])t([0-2])$
---
> ^x([0-9]{1,4})c([0-7])t([0-9])$
15a19,20
> ^x([0-9]{1,4})c([0-7])s([0-9]+)b([0-9]+)n([0-9]+)g([0-9]+)$
> ^x([0-9]{1,4})c([0-7])s([0-9]+)b([0-9]+)n([0-9]+)g([0-9]+)k([0-9]+)$
20c25,26
< ^x([0-9]{1,4})c([0-7])s([0-9]+)j([1-2])$
---
> ^x([0-9]{1,4})c([0-7])s([0-9]+)e([0-9]+)t([0-9]+)$
> ^x([0-9]{1,4})c([0-7])s([0-9]+)[jv]([1-2])$
22a29
> ^x([0-9]{1,4})c([0-7])s([0-9]+)b([0-9]+)n([0-9]+)v([0-9]+)$
25c32,33
< ^x([0-9]{1,4})c([0-7])s([0-9]+)b([0-9]+)n([0-9]+)a([0-7])$
---
> ^x([0-9]{1,4})c([0-7])s([0-9]+)b([0-9]+)n([0-9]+)a([0-9]+)$
> ^x([0-9]{1,4})c([0-7])s([0-9]+)b([0-9]+)n([0-9]+)r([0-7])$
29a38
> ^x([0-9]{1,4})c([0-7])r([0-9]+)t([0-9]+)$
32a42
> ^x([0-9]{1,4})c([0-7])r([0-9]+)v([1-2])$
38c48,50
< ^x([0-9]{1,4})c([0-7])w([0-9]+)j([1-9][0-9]*)$
---
> ^x([0-9]{1,4})c([0-7])w([1-9][0-9]*)j([1-9][0-9]*)$
> ^x([0-9]{1,4})c([0-7])h([1-9][0-9]*)$
> ^x([0-9]{1,4})c([0-7])h([1-9][0-9]*)s([1-9])$
```

**Comment by shunr-hpe** [comment]
Here is the diff for the `hwtype_pattern` changes
```
> ^CDUMgmtSwitch$
8a10
> ^CabinetPDUPowerConnector$
10a13
> ^ChassisBMCNic$
18a22
> ^NodeEnclosurePowerSupply$
23,24c27,30
< ^NodeNIC$
< ^NodeHsnNIC$
---
> ^StorageGroup$
> ^Drive$
> ^NodeNic$
> ^NodeHsnNic$
26a33
> ^NodeAccelRiser$
29a37
> ^RouterTOR$
32a41
> ^RouterPowerConnector$
37a47,48
> ^MgmtHLSwitchEnclosure$
> ^MgmtHLSwitch$
```

File: test/ct/api/2-disruptive/test_hardware.tavern.yaml (hunk 1 above)
Comment (shunr-hpe) [reply test/ct/api/2-disruptive/test_hardware.tavern.yaml:279]: I added a test with post

**Review by rsjostrand-hpe — APPROVED** [review]

**Review by mjendrysik-hpe — APPROVED** [review]

**Pull request merged**